src/mme-app/actionHandlers/attachActionHandlers.cpp

```diff
@@ -1021,3 +1030,42 @@ ActStatus ActionHandlers::attach_done(SM::ControlBlock& cb)
 
 	return ActStatus::PROCEED;
 }
+
+ActStatus ActionHandlers::send_attach_reject(ControlBlock& cb)
+{
+        UEContext* ueCtxt_p = static_cast<UEContext*>(cb.getPermDataBlock());
+        if (ueCtxt_p == NULL)
+        {
+                log_msg(LOG_ERROR, " send_attach_reject: UE context is NULL %d\n",cb.getCBIndex());
+                return ActStatus::HALT;
+        }
+
+        MmeProcedureCtxt *procCtxt = dynamic_cast<MmeProcedureCtxt*>(cb.getTempDataBlock());
+        if (procCtxt == NULL)
+        {
+                log_msg(LOG_DEBUG, "send_attach_reject: Procedure context is NULL\n");
+                return ActStatus::HALT;
+        }
+        
+        struct commonRej_info attach_rej;
```

Comment:
Please improve this to move some of the code in constructor ?

Reply:
These are shared struct between c and c++ code. Will check how the init can be optimized.

Reply:
Created task for this.